vrt/ci.js

```diff
@@ -0,0 +1,351 @@
+  try {
+    await octokit.pulls.update({
+      owner: `uber`,
+      repo: `baseweb`,
+      pull_number: snapshotPullRequestNumber,
+      state: 'closed',
+    });
+    _echo(`Closed the existing snapshot PR.`);
+  } catch (er) {
+    _echo(`There was an error closing the existing snapshot PR.`);
+    _echo(er);
+  }
+  await notifyOriginalPullRequestOfClosure(snapshotPullRequestNumber);
+}
+
+async function notifyOriginalPullRequestOfClosure(snapshotPullRequestNumber) {
+  try {
+    await octokit.issues.createComment({
+      owner: `uber`,
+      repo: `baseweb`,
+      issue_number: BUILDKITE_PULL_REQUEST,
+      body:
+        `Visual changes have been resolved. #${snapshotPullRequestNumber} has been closed. ` +
+        `A new snapshot branch will be created and a new PR will be opened if future commits on \`${BUILDKITE_BRANCH}\` trigger visual changes.`,
+    });
+    _echo(
+      `Posted a comment on original PR about closure of existing snapshot PR.`,
+    );
+  } catch (er) {
+    _echo(
+      `There was an error commenting on the original PR about snapshot resolution.`,
+    );
+    _echo(er);
+  }
+}
+
+async function notifySnapshotPullRequestOfClosure(snapshotPullRequestNumber) {
+  try {
+    await octokit.issues.createComment({
+      owner: `uber`,
+      repo: `baseweb`,
+      issue_number: snapshotPullRequestNumber,
+      body:
+        `Visual changes have been resolved. ` +
+        `This PR will be closed and \`${SNAPSHOT_BRANCH}\` will be deleted. ` +
+        `A new snapshot branch will be created and a new PR will be opened if future commits on \`${BUILDKITE_BRANCH}\` trigger visual changes.`,
+    });
+    _echo(
+      `Posted a comment on snapshot PR about visual resolution and closure.`,
+    );
+  } catch (er) {
+    _echo(
+      `There was an error commenting on the existing snapshot PR about snapshot resolution.`,
+    );
+    _echo(er);
+  }
+}
+
+async function addLabelsToNewPullRequest(newPullRequestNumber) {
+  try {
+    await octokit.issues.addLabels({
+      owner: `uber`,
+      repo: `baseweb`,
+      issue_number: newPullRequestNumber,
+      labels: [`bugfix`, `ci`],
+    });
+    _echo(`Added labels to new snapshot PR.`);
+  } catch (er) {
+    _echo(`There was an error adding labels to new snapshot PR.`);
+    _echo(er);
+  }
+}
+
+async function addCommentToOriginalPullRequest(newPullRequestNumber) {
+  try {
+    const comment = await octokit.issues.createComment({
+      owner: `uber`,
+      repo: `baseweb`,
+      issue_number: BUILDKITE_PULL_REQUEST,
+      body:
+        `Visual changes were detected on this branch. ` +
+        `Please review the following PR containing updated snapshots: #${newPullRequestNumber}`,
+    });
+    _echo(
+      `Posted a comment linking to snapshot PR on original PR: ${comment.data.html_url}`,
+    );
+  } catch (er) {
+    _echo(`Error creating comment on original PR.`);
+    _echo(er);
+  }
+}
+
+async function openNewSnapshotPullRequest() {
+  try {
```

Comment:
I would remove the the try/catch blocks, and let the error propagate back to the main function - that way you only need to do error handling once

I'd do the same for all the utility funcitons

Reply:
so you wouldn't have to do the process.exits either